src/main/java/com/mongodb/jdbc/ExtJsonValue.java

```diff
@@ -0,0 +1,109 @@
+
+import java.io.StringWriter;
+import java.sql.Date;
+import java.text.SimpleDateFormat;
+import org.bson.BsonArray;
+import org.bson.BsonValue;
+import org.bson.codecs.BsonValueCodecProvider;
+import org.bson.codecs.Codec;
+import org.bson.codecs.EncoderContext;
+import org.bson.codecs.configuration.CodecRegistry;
+import org.bson.json.JsonMode;
+import org.bson.json.JsonWriterSettings;
+
+/**
+ * ExtJsonValue is a wrapper for BsonValue. The purpose of this class is to override the toString()
+ * method to produce the extended JSON representation of a BsonValue, rather than the java driver's
+ * default BsonValue.toString() output.
+ *
+ * <p>The driver's BsonValue class is abstract and intentionally cannot be extended by third
+ * parties. The driver explains this is to keep the BSON type system closed.
+ */
+public class ExtJsonValue {
+    static final JsonWriterSettings JSON_WRITER_SETTINGS =
+            JsonWriterSettings.builder().outputMode(JsonMode.RELAXED).build();
+    static final CodecRegistry CODEC_REGISTRY = fromProviders(new BsonValueCodecProvider());
+    static final EncoderContext ENCODER_CONTEXT = EncoderContext.builder().build();
+
+    // dateFormat cannot be static due to a threading bug in the library.
+    protected SimpleDateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSS'Z'");
+
+    private BsonValue v;
+
+    public ExtJsonValue(BsonValue v) {
+        this.v = v;
+    }
+
+    /** @return The underlying BsonValue */
+    public BsonValue getBsonValue() {
+        return this.v;
+    }
+
+    @Override
+    public String toString() {
+        switch (this.v.getBsonType()) {
+            case ARRAY:
+                Codec<BsonArray> codec = CODEC_REGISTRY.get(BsonArray.class);
+                StringWriter writer = new StringWriter();
+                codec.encode(
+                        new NoCheckStateJsonWriter(writer, JSON_WRITER_SETTINGS),
+                        this.v.asArray(),
+                        ENCODER_CONTEXT);
+                writer.flush();
+                return writer.toString();
+            case BOOLEAN:
+                return this.v.asBoolean().getValue() ? "true" : "false";
+            case DATE_TIME:
+                Date d = new Date(this.v.asDateTime().getValue());
+                return dateFormat.format(d);
+            case DECIMAL128:
+                return this.v.asDecimal128().getValue().toString();
+            case DOCUMENT:
+                return this.v.asDocument().toJson(JSON_WRITER_SETTINGS);
+            case DOUBLE:
+                return Double.toString(this.v.asDouble().getValue());
+            case INT32:
+                return Integer.toString(this.v.asInt32().getValue());
+            case INT64:
+                return Long.toString(this.v.asInt64().getValue());
+            case NULL:
+                return null;
+            case OBJECT_ID:
+                return this.v.asObjectId().getValue().toString();
```

Comment:
I did a run a quick test and BsonObjectId.getValue().toString() does not return the relaxed extended Json representation.
BsonObjectId.getValue().toString() => 62478aa9cc79dc251e626f7d
Relaxed extended JSON representation (Same a canonical for oid) => {"$oid": "62478aa9cc79dc251e626f7d"}

Do we want a special treatment for OIDs? IMO the extended JSON representation would be good there because it makes it clear its not a simple string. I thought this was something we wanted to aim for, but I remember OIDs are a bit of a muddy ground so maybe I missed something.

Reply:
This was initially implemented before we decided this ticket would cover all types (not just object and array) so it does not have the work we discussed in standup on friday! will work on that today. @bucaojit I like that idea, but I will file a follow-up ticket to introduce that.

Reply:
done:
1. PR updated to have extended json stringification for all types.OTHER types, and
2. filed [SQL-813](https://jira.mongodb.org/browse/SQL-813) to follow-up on Oliver's suggestion